src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web.UnitTests/Controllers/ManageApprenticesTests/WhenIGoToManageMyApprentices.cs

```diff
@@ -1,42 +1,105 @@
-﻿using Microsoft.AspNetCore.Mvc;
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoFixture;
+using AutoFixture.NUnit3;
+using Microsoft.AspNetCore.Mvc;
+using Moq;
 using NUnit.Framework;
+using SFA.DAS.Commitments.Shared.Interfaces;
+using SFA.DAS.CommitmentsV2.Types;
 using SFA.DAS.ProviderCommitments.Web.Controllers;
 using SFA.DAS.ProviderCommitments.Web.Models;
+using SFA.DAS.Testing.AutoFixture;
 
 namespace SFA.DAS.ProviderCommitments.Web.UnitTests.Controllers.ManageApprenticesTests
 {
     [TestFixture]
     public class WhenIGoToManageMyApprentices
     {
-        private ManageApprenticesController _controller;
+        [Test, MoqAutoData]
+        public void IfCalledWithAnInvalidRequestShouldGetBadResponseReturned(
+            [Frozen]Mock<ICommitmentsService> commitmentsService,
+            ManageApprenticesController controller)
+        {
+            //Arrange
+            controller.ModelState.AddModelError("test", "test");
+
+            //Act
+            var result = controller.Index(1);
+
+            //Assert
+            Assert.IsInstanceOf<BadRequestObjectResult>(result.Result);
+        }
 
-        [SetUp]
-        public void Arrange()
+        [Test, MoqAutoData]
+        public void ThenTheProviderIdIsPassedToTheViewModel(
+            [Frozen]Mock<ICommitmentsService> commitmentsService,
+            ManageApprenticesController controller,
+            int providerId)
         {
-            _controller = new ManageApprenticesController();
+            //Arrange
+            
+            //Act
+            var result = controller.Index(providerId);
+
+            var view = ((ViewResult) result.Result).Model as ManageApprenticesViewModel;
+
+            //Assert
+            Assert.AreEqual(view.ProviderId, providerId);
         }
 
-        [Test]
-        public void IfCalledWithAnInvalidRequestShouldGetBadResponseReturned()
+        [Test, MoqAutoData]
+        public void ThenTheApprovedApprenticesAreSet(
+            [Frozen]Mock<ICommitmentsService> commitmentsService,
+            ManageApprenticesController controller,
+            List<ApprenticeshipDetails> approvedApprenticeships)
         {
             //Arrange
-            _controller.ModelState.AddModelError("test", "test");
+            commitmentsService.Setup(x => x.GetApprovedApprenticeships(It.IsAny<uint>()))
+                .ReturnsAsync(approvedApprenticeships);
+
             //Act
-            var result = _controller.Index(1);
+            var result = controller.Index(1);
+            var view = ((ViewResult)result.Result).Model as ManageApprenticesViewModel;
+
             //Assert
-            Assert.IsInstanceOf<BadRequestObjectResult>(result);
+            Assert.AreEqual(view.Apprenticeships.Count(), approvedApprenticeships.Count);
         }
 
+/*        [TestCase(0, false)]
+        [TestCase(1, true)]
+        [TestCase(10, true)]*/
```

Comment:
Given what this test is testing for i'd either use 2 rather than 10 or just remove it altogether. The reason why 2 would be better as it will take less time to complete and still test the many aspect of the test data.